service/geopmdpy/system_files.py

```diff
@@ -270,6 +271,42 @@ def is_secure_file(path, fid):
     return result
 
 
+def check_plugins_secure():
+    """Verify custom plugins are secure before use
+
+    Check to ensure all custom plugins that may be used by the runtime
+    can be securely read before use.  After this has been called, plugin
+    files that do not meet the permissions requirements will be renamed
+    such that they are ignored by the runtime.  Files that pass the check
+    will be loaded by the runtime.
+
+    """
+    plugin_path = os.getenv('GEOPM_PLUGIN_PATH')
+    if plugin_path is not None:
+        user_paths = plugin_path.split(':')
+        for path in user_paths:
+            secure_make_dirs(path)
+            scanned_paths = set()
+            # Custom IOGroup/Agent plugins: geopm_plugin.cpp -> plugin_load()
+            for file_path in glob.glob(os.path.join(path, 'lib*.so*')):
+                if not file_path.endswith('INVALID'):
+                    with open(file_path, 'r') as ff:
+                        scanned_paths.add(file_path)
+                        if is_secure_file(file_path, ff):
+                            so_suffix = '.so.' + version.__abi_version__.replace(':', '.')
```

Comment:
I guess these file checks are better than not having them, but there is a TOCTOU issue with this solution.